addons/event_sale/static/src/client_action/event_registration_summary_dialog.js

```diff
@@ -0,0 +1,12 @@
+import { patch } from "@web/core/utils/patch";
+import { EventRegistrationSummaryDialog } from "@event/client_action/event_registration_summary_dialog";
+
+patch(EventRegistrationSummaryDialog.prototype, {
+    setup() {
+        super.setup(...arguments);
+        if (this.registration.has_to_pay) {
+            this.willAutoPrint = false;
+            this.dialogState.isHidden = false;
```

Comment:
Would be slightly cleaner to make "willAutoPrint" a getter, that way you can override it here without having to re-assign the dialogState.

Not a big deal though, but as we're cleaning this...